src/main/transports/offline-store.ts

```diff
@@ -63,7 +63,7 @@ export function createOfflineStore(userOptions: Partial<OfflineStoreOptions>): O
     });
   }
 
-  function removeStaleRequests(queue: PersistedRequest[]): void {
+  function removeStaleRequests(queue: Array<Partial<PersistedRequest> | undefined>): void {
     while (queue[0] && isOutdated(queue[0], options.maxAgeDays)) {
```

Comment:
given we do the `queue[0]` check here, can we enforce that `isOutdated` gets `Partial<PersistedRequest>` instead of `Partial<PersistedRequest> | undefined`?